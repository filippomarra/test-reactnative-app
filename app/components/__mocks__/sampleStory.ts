import Story from '../../models';

export const sampleStory: Story = {
    kind: 'story',
    id: 151723133,
    created_at: new Date('2017-10-05T15:30:45Z'),
    updated_at: new Date('2020-09-09T08:08:04Z'),
    accepted_at: new Date('2017-10-18T14:04:39Z'),
    story_type: 'chore',
    story_priority: 'p3',
    name: 'API Design',
    description:
        'As an engineer, I will produce a RAML file to discuss with the team the entry points and the payloads for the proxy so that we can start thinking how to mock this data in the frontend',
    current_state: 'accepted',
    requested_by_id: 1945893,
    url: 'https://www.pivotaltracker.com/story/show/151723133',
    project_id: 2116794,
    owner_ids: [2958113],
    labels: [
        {
            id: 19223788,
            project_id: 2116794,
            kind: 'label',
            name: 'debito-tecnico',
            created_at: new Date('2017-10-05T15:30:45Z'),
            updated_at: new Date('2020-09-09T08:08:04Z'),
        },
    ],
    owned_by_id: 2958113,
};
