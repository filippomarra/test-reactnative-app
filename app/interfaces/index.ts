import Story from '../models';

export interface ApiProps {
    setLoading: (loading: boolean) => void;
    setStories: (stories: Story[]) => void;
}

export interface AppHeaderProps {
    stories: Story[];
    setStories: (stories: Story[]) => void;
}

export interface ModalProps {
    visible: boolean;
    hideDialog: () => void;
    story: Story;
}

export interface StoryItemProps {
    index: number;
    showDialog: (story: Story) => void;
    story: Story;
}

export interface StoryScrollViewProps {
    isLoading: boolean;
    showDialog: (story: Story) => void;
    stories: Story[];
}
