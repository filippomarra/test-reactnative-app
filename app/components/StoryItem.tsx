import React from 'react';
import { View } from 'react-native';
import { Divider, ListItem } from '@rneui/themed';

import { StoryItemProps } from '../interfaces';
import { styles } from '../styles';

export const StoryItem = (props: StoryItemProps) => {
    const { index, showDialog, story } = props;
    return (
        <View>
            <ListItem
                key={index}
                onPress={() => showDialog(story)}
                testID="storyItem">
                <ListItem.Content>
                    <ListItem.Title style={styles.title}>
                        {story.name}
                    </ListItem.Title>
                    <ListItem.Subtitle style={styles.subtitle}>
                        {story.description && story.description.length > 70
                            ? story.description.substring(0, 70) + '...' // Truncate long descriptions
                            : story.description}
                    </ListItem.Subtitle>
                </ListItem.Content>
                <ListItem.Chevron size={30} color="#000" />
            </ListItem>
            <Divider color="#000" />
        </View>
    );
};
