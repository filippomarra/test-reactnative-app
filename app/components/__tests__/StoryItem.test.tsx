import 'react-native';
import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import { StoryItemProps } from '../../interfaces';

import { sampleStory } from '../__mocks__/sampleStory';
import { StoryItem } from '../StoryItem';

const defaultProps: StoryItemProps = {
    index: 0,
    showDialog: jest.fn(),
    story: {},
};

describe('StoryItem component tests', () => {
    it('renders correctly with default props', () => {
        expect(
            render(<StoryItem {...defaultProps} />).toJSON(),
        ).toMatchSnapshot();
    });

    describe('When the story item is not pressed', () => {
        it('should not invoke the setStories function', () => {
            const showDialogSpy = jest.fn();
            render(<StoryItem {...defaultProps} showDialog={showDialogSpy} />);
            expect(showDialogSpy).not.toHaveBeenCalled();
        });
    });

    describe('When the story item is pressed', () => {
        it('should invoke the setStories function', () => {
            const showDialogSpy = jest.fn();
            const { getByTestId } = render(
                <StoryItem {...defaultProps} showDialog={showDialogSpy} />,
            );
            const storyItem = getByTestId('storyItem');
            fireEvent(storyItem, 'onPress');
            expect(showDialogSpy).toHaveBeenCalled();
        });
    });
});
