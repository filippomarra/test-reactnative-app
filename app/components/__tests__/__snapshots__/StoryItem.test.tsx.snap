// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryItem component tests renders correctly with default props 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="storyItem"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        theme={
          Object {
            "colors": Object {
              "background": "#ffffff",
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#03dac4",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
            "spacing": Object {
              "lg": 12,
              "md": 8,
              "sm": 4,
              "xl": 24,
              "xs": 2,
            },
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#242424",
              "fontFamily": "TitilliumWeb-SemiBold",
              "fontSize": 24,
            }
          }
          testID="listItemTitle"
        />
        <Text
          accessibilityRole="text"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#5C6784",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        />
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "overflow": "hidden",
          }
        }
        testID="RNE__ICON__CONTAINER"
      >
        <View
          accessibilityRole="button"
          android_ripple={
            Object {
              "borderless": false,
              "color": "rgba(0, 0, 0, 0.3)",
              "radius": -5,
            }
          }
          disabled={false}
          testID="RNE__ICON__CONTAINER_ACTION"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            testID="RNE__ICON"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="RNE__ICON__Component"
            >
              ïˆ˜
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#000",
        "borderBottomWidth": 0.5,
      }
    }
    testID="RNE__Divider"
    theme={
      Object {
        "colors": Object {
          "background": "#ffffff",
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#03dac4",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
        "spacing": Object {
          "lg": 12,
          "md": 8,
          "sm": 4,
          "xl": 24,
          "xs": 2,
        },
      }
    }
  />
</View>
`;
